go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -289,31 +356,55 @@ func (t *txThrottler) Close() {
 	log.Info("txThrottler: closed")
 }
 
-// Throttle should be called before a new transaction is started.
-// It returns true if the transaction should not proceed (the caller
-// should back off). Throttle requires that Open() was previously called
-// successfully.
-func (t *txThrottler) Throttle(priority int, workload string) (result bool) {
+// Throttle should be called before a new transaction is started. It
+// returns an error if the transaction should not proceed (the caller
+// should back off). Throttle requires that Open() was previously
+// called successfully.
+func (t *txThrottler) Throttle(plan *planbuilder.Plan, options *querypb.ExecuteOptions) (err error) {
 	if !t.config.enabled {
-		return false
+		return nil
 	}
 	if t.state == nil {
-		return false
+		return nil
 	}
 
-	// Throttle according to both what the throttler state says and the priority. Workloads with lower priority value
-	// are less likely to be throttled.
-	result = t.state.throttle() && rand.Intn(sqlparser.MaxPriorityValue) < priority
+	// get priority from execute options, skip if the priority
+	// is equal to sqlparser.MinPriorityValue
+	planType := plan.PlanID.String()
+	metricLabels := []string{planType, options.GetWorkloadName()}
```

Comment:
Do we really need the plan type in the metrics? What use case do you see for it?